### PR TITLE
Feat/snyk fix requirements txt support v1

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Introduce the v1 functionality of @snyk/fix package that includes:
- Detecting and forwarding test results to relevant plugin for fixing
- Basic Python support for `requirements.txt` projects & their custom name variations
- Skip any projects that are not supported (e.g. npm, maven, requirements files with `-r` or `0c` directive, projects with no `remediation` data object)

#### Where should the reviewer start?
`packages/snyk-fix/src/index.ts`

#### How should this be manually tested?
`npm run test` Integration with CLI is not yet in place.

#### Screenshots
CleanShot 2021-03-05 at 12 32 12@2x

### CODE REVIEW

File: packages/snyk-fix/src/lib/errors/custom-error.ts [hunk 1]
```diff
@@ -11,5 +11,7 @@ export class CustomError extends Error {
 }
 
 export enum ERROR_CODES {
-  UnsupportedTypeError = 'G1',
+  UnsupportedTypeError = 'G10',
```
Comment (lili2311) [inline packages/snyk-fix/src/lib/errors/custom-error.ts:14]: some codes for now, we have a session planned to look at these and decide on a common way forward

File: packages/snyk-fix/src/plugins/types.ts [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
+export interface PluginFixResponse {
+  succeeded: Array<WithFixChangesApplied<EntityToFix>>;
+  failed: Array<WithError<EntityToFix>>;
+  skipped: Array<WithUserMessage<EntityToFix>>;
 }
+export interface FixHandlerResultByPlugin {
+  [pluginId: string]: PluginFixResponse;
+}
+// ecosystem is the plugin
+export type Ecosystem = 'python';
```
Comment (aarlaud) [inline packages/snyk-fix/src/plugins/types.ts:23]: Why is there this hardcoded python value here? Isn't this file meant to be a type definition set for the plugins?

File: packages/snyk-fix/test/acceptance/plugins/python/update-dependencies/__snapshots__/update-dependencies.spec.ts.snap [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (aarlaud) [inline packages/snyk-fix/test/acceptance/plugins/python/update-dependencies/__snapshots__/update-dependencies.spec.ts.snap:1]: What is this file?

**Review by aarlaud — APPROVED** [review]

File: packages/snyk-fix/test/acceptance/plugins/python/update-dependencies/__snapshots__/update-dependencies.spec.ts.snap (hunk 3 above)
Comment (lili2311) [reply packages/snyk-fix/test/acceptance/plugins/python/update-dependencies/__snapshots__/update-dependencies.spec.ts.snap:1]: this is a test snapshot, so when you use `expect(x).toMatchSnaphot()` jest writes ones of these. So if anything changes you will see a diff and test fails. https://jestjs.io/docs/en/snapshot-testing

File: packages/snyk-fix/src/plugins/types.ts (hunk 2 above)
Comment (lili2311) [reply packages/snyk-fix/src/plugins/types.ts:23]: good find, it is not used anywhere. Removing

**Pull request merged**